src/Query/Providers/IQueryProvider.cs

```diff
@@ -164,6 +164,17 @@ public interface IQueryProvider : System.Linq.IQueryProvider, IDisposable
         /// </returns>
         Task<int> ExecuteNonQueryAsync(string query);
 
+        /// <summary>
+        /// Asynchronously executes arbitrary query.
+        /// </summary>
+        /// <param name="query">Arbitrary SQL query. It should not be SELECT query.</param>
+        /// <param name="parameters">List of query parameters.</param>
+        /// <returns>
+        /// A task that represents the asynchronous operation. The task result contains the
+        /// numbers of affected rows.
+        /// </returns>
+        Task<int> ExecuteNonQueryAsync(string query, params object[] parameters);
```

Comment:
I would rename `parameters` to `values` (or `paramValues` or something similar) - they really are values. 😉